From ff78a020969e87c5a108008fc3fa9940b99a3856 Mon Sep 17 00:00:00 2001
From: vpalvai <vpalvai@codeaurora.org>
Date: Mon, 14 Aug 2017 11:27:18 -0700
Subject: [PATCH 2/2] ComputeLibrary: add absolute path for cl2.hpp

 - provide absolute path for cl2.hpp, as this
   file is not yet available on AGL builds
---
 CMakeLists.txt               | 1 +
 arm_compute/core/CL/OpenCL.h | 4 +++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 012922e..b30bdb0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,6 +78,7 @@ endif (ACL_NEON)
 ADD_LIBRARY (${TARGET_NAME} SHARED ${core_files} ${runtime_files})
 INSTALL (TARGETS ${TARGET_NAME} DESTINATION lib64)
 INSTALL (DIRECTORY arm_compute DESTINATION include)
+INSTALL (FILES "include/CL/cl2.hpp" DESTINATION include)
 
 add_executable (neon_convolution examples/neon_convolution.cpp test_helpers/Utils.cpp)
 target_link_libraries(neon_convolution arm_compute OpenCL)
diff --git a/arm_compute/core/CL/OpenCL.h b/arm_compute/core/CL/OpenCL.h
index 9d39e5d..9457e81 100644
--- a/arm_compute/core/CL/OpenCL.h
+++ b/arm_compute/core/CL/OpenCL.h
@@ -29,7 +29,9 @@
 #define CL_HPP_CL_1_2_DEFAULT_BUILD
 #define CL_HPP_TARGET_OPENCL_VERSION 110
 #define CL_HPP_MINIMUM_OPENCL_VERSION 110
-#include <CL/cl2.hpp>
+// provide absolute path for cl2.hpp, as this
+// file is not yet available on AGL builds
+#include "../../../include/CL/cl2.hpp"
 
 namespace cl
 {
-- 
2.13.0

